### PR TITLE
Adding ToMe parameters to available x/y/z plot options

### PR DESCRIPTION
**Pull request by nekoworkshop**
This addresses 

- https://github.com/vladmandic/automatic/issues/532 
- a bug regarding the component for selecting maximum down sampling in the settings page https://github.com/vladmandic/automatic/issues/532#issuecomment-1524044984 

In addition, I have added some extra information to some ToMe settings based on the [original plugin author's comments](https://github.com/dbolya/tomesd/blob/2e37b87f2f24b0c902507b8afb7ebc7ba9b7cf6b/tomesd/patch.py#L196).  This should help the users get started with ToMe easier and preventing them from shooting their own feet (Enabling merge_mlp results in greatly reduced output quality). 

I have tested my change locally after rebasing to 6cd4d62f710a5c49dc6ff46a5dfb9ca93c8c5246

xyz_grid-0241-3184478247-DPM++ 2M Karras-20-7-20230429184205

### CODE REVIEW

File: modules/shared.py [hunk 1]
```diff
@@ -422,14 +422,14 @@ def refresh_themes():
...
     "token_merging_random": OptionInfo(False, "Use random perturbations - Can improve outputs for certain samplers. For others, it may cause visual artifacting.", gr.Checkbox),
-    "token_merging_merge_attention": OptionInfo(True, "Merge attention", gr.Checkbox),
-    "token_merging_merge_cross_attention": OptionInfo(False, "Merge cross attention", gr.Checkbox),
-    "token_merging_merge_mlp": OptionInfo(False, "Merge mlp", gr.Checkbox),
-    "token_merging_maximum_down_sampling": OptionInfo(1, "Maximum down sampling", gr.Dropdown, lambda: {"choices": ["1", "2", "4", "8"]}),
+    "token_merging_merge_attention": OptionInfo(True, "Merge attention (Recommend on)", gr.Checkbox),
+    "token_merging_merge_cross_attention": OptionInfo(False, "Merge cross attention (Recommend off)", gr.Checkbox),
+    "token_merging_merge_mlp": OptionInfo(False, "Merge mlp (Strongly recommend off)", gr.Checkbox),
+    "token_merging_maximum_down_sampling": OptionInfo(1, "Maximum down sampling", gr.Radio, lambda: {"choices": [1, 2, 4, 8]}),
```
Comment (nekoworkshop) [inline modules/shared.py:432]: Fix for https://github.com/vladmandic/automatic/issues/532#issuecomment-1524044984

**Comment by vladmandic** [comment]
thanks.

only thing that soft of bothers me is how number of params in xyz scripts is starting to grow out of control.
controlnet added 10 params, now tome is adding 10 more, etc. - i'm all for tome in xyz script, but lets keep it to absolute minimum. just as an example, does anyone other than author of tome actually need "ToMe merge mlp" in xyz script?

similar, but less extreme - if something is "strongly recommended off", lets just assume that it is off and not present option. i don't want to follow a1111 path where there are thousands of options that nobody knows what they do.

btw, i just spend an hour today removing items from settings :)

**Comment by nekoworkshop** [comment]
@vladmandic Actually I was hoping you would say that so I could have the excuse not to include all of them😄. Should have checked with you sooner I guess.

Personally I would only keep the following

- ToMe ratio
- ToMe ratio for hires fix
- ToMe random pertubations

The ratios are _the_ parameter to control the balance between quality and speed/size. From my personal experience Random pertubations could be model/sampler/prompt dependent. I would keep it so the users can find out whether they should leave it on or off easier.  The author suggests leaving it open but turn it off if it causes artifacts https://github.com/dbolya/tomesd/blob/2e37b87f2f24b0c902507b8afb7ebc7ba9b7cf6b/tomesd/patch.py#LL208C17-L208C17

**Comment by vladmandic** [comment]
perfect.
what's the value of having different ratios for standard vs hires fix steps?

**Comment by nekoworkshop** [comment]
@vladmandic I'm running a 3060 so I'm not too bothered by the 1st pass generation speed, but for people with less power rigs they are likely going to use ToMe for both 1st pass and Hires pass. I would say offering the option for both passes can give them the possibility to find out a good combination of rations for both pass.

I'm going to remove the verbose ToMe options from the x/y/z plot then update my PR.

For the settings page I think we can keep it as it is for now since those options are self-contained in the ToMe tab. If someone really wants to play with those knobs and switches it's their choice😅

**Comment by vladmandic** [comment]
sounds good

**Pull request merged**